src/components/Swap/Toolbar/index.tsx

```diff
@@ -186,7 +171,17 @@ export default memo(function Toolbar() {
   return (
     <StyledExpando
       title={
-        <ToolbarRow flex justify="space-between" data-testid="toolbar">
+        <ToolbarRow
+          flex
+          justify="space-between"
+          data-testid="toolbar"
+          onClick={(e: MouseEvent<HTMLDivElement>) => {
+            const el = e.target as HTMLDivElement
+            if (!el.classList.contains(TradePriceToggledClass)) {
+              setOpen((open) => !open)
+            }
+          }}
+        >
```

Comment:
i'd define this function up in the component for readability

Reply:
and then also do the guard-style `if` so like:

```
// Comment here about preventing Toolbar from opening when clicking on the Trade price component
if (el.classList.contains(TradePriceToggledClass)) return

setOpen(open => !open)
```

Reply:
i avoid if statements without braces, so my change is slightly different than the suggestion.